src/CorrectPlayerMovePredictionPacket.php

```diff
@@ -52,20 +59,33 @@ public function getTick() : int{ return $this->tick; }
 
 	public function getPredictionType() : int{ return $this->predictionType; }
 
+	public function getVehicleRotation() : ?Vector2{ return $this->vehicleRotation; }
+
 	protected function decodePayload(PacketSerializer $in) : void{
+		$this->predictionType = $in->getByte();
 		$this->position = $in->getVector3();
 		$this->delta = $in->getVector3();
+		if($this->predictionType === self::PREDICTION_TYPE_VEHICLE) {
+			$this->vehicleRotation = new Vector2($in->getFloat(), $in->getFloat());
+		}
 		$this->onGround = $in->getBool();
 		$this->tick = $in->getUnsignedVarLong();
-		$this->predictionType = $in->getByte();
 	}
 
 	protected function encodePayload(PacketSerializer $out) : void{
+		$out->putByte($this->predictionType);
 		$out->putVector3($this->position);
 		$out->putVector3($this->delta);
+		if($this->predictionType === self::PREDICTION_TYPE_VEHICLE) {
+			if($this->vehicleRotation == null) { // this should never be the case
+				throw new \InvalidArgumentException("CorrectPlayerMovePredictionPackets with type VEHICLE require a vehicleRotation to be provided");
```

Comment:
This can be a LogicException, since it should've been checked by the constructor